python_modules/libraries/dagster-dbt/dagster_dbt/cli/resources_v2.py

```diff
@@ -130,6 +136,67 @@ def get_asset_key_by_output_name(self, output_name: str) -> AssetKey:
         """Get a dbt node's default asset key by its Dagster output name."""
         return self.node_info_to_asset_key(self.get_node_info_by_output_name(output_name))
 
+    def get_asset_key_for_dbt_unique_id(self, unique_id: str) -> AssetKey:
+        node_info = self.node_info_by_dbt_unique_id.get(unique_id)
+
+        if not node_info:
+            raise DagsterInvalidInvocationError(
+                f"Could not find a dbt node with unique_id: {unique_id}. A unique ID consists of"
+                " the node type (model, source, seed, etc.), project name, and node name in a"
+                " dot-separated string. For example: model.my_project.my_model\n For more"
+                " information on the unique ID structure:"
+                " https://docs.getdbt.com/reference/artifacts/manifest-json"
+            )
+
+        return self.node_info_to_asset_key(node_info)
+
+    def get_asset_keys_by_output_name_for_source(self, source_name: str) -> Mapping[str, AssetKey]:
+        """For a dbt source, returns a mapping of output name (stringified from the dbt unique
+        ID) to dagster asset key for each table in the source.
+        """
+        check.str_param(source_name, "source_name")
+
+        matching_nodes = [
+            value
+            for value in self.raw_manifest["sources"].values()
+            if value["source_name"] == source_name
+        ]
+
+        if len(matching_nodes) == 0:
+            raise DagsterInvalidInvocationError(
+                f"Could not find a dbt source with name: {source_name}"
+            )
+
+        return {
+            output_name_fn(value): self.node_info_to_asset_key(value) for value in matching_nodes
+        }
+
+    def get_asset_key_for_model(
+        self, model_name: str, key_prefix: Optional[CoercibleToAssetKeyPrefix] = None
```

Comment:
Not a fan of this `key_prefix` argument... but its existence precedes this PR

Reply:
We can remove the `key_prefix` arg. Instead, users can just use `AssetKey(...).with_prefix(..)` instead